torchani/data/new.py

```diff
@@ -0,0 +1,449 @@
+import numpy as np
+import torch
+import functools
+from ._pyanitools import anidataloader
+import importlib
+pkbar_installed = importlib.util.find_spec('pkbar')
+if pkbar_installed is not None:
+    import pkbar
+
+
+class CacheDataset(torch.utils.data.Dataset):
+
+    def __init__(self, file_path,
+                 species_order='HCNO',
+                 transform=True,
+                 self_energies=[-0.600953, -38.08316, -54.707756, -75.194466],
+                 batch_size=1000,
+                 device='cpu',
+                 bar=20,
+                 test_bar_max=None):
+
+        super(CacheDataset, self).__init__()
+
+        # species_dict = {'H': 0, 'C': 1, 'N': 2, 'O': 3}
+        # self_energies_dict = {'H': -0.600953, 'C': -38.08316, 'N': -54.707756, 'O': -75.194466}
+        species_dict = {}
+        self_energies_dict = {}
+        for i, s in enumerate(species_order):
+            species_dict[s] = i
+            self_energies_dict[s] = self_energies[i]
+
+        self.data_species = []
+        self.data_coordinates = []
+        self.data_energies = []
+        self.data_self_energies = []
+
+        anidata = anidataloader(file_path)
+        anidata_size = anidata.group_size()
+        if anidata_size > 5 and pkbar_installed is not None:
```

Comment:
The check of the same condition appears many times, please change it to
```python
enable_pkbar = anidata_size > 5 and PKBAR_INSTALLED
if enable_pkbar:
    ...
```

Reply:
only two times for each class, I don't think it's necessary